packages/bsky/src/api/app/bsky/graph/getList.ts

```diff
@@ -68,7 +70,31 @@ const hydration = async (
       params.hydrateCtx,
     ),
   ])
-  return mergeStates(listState, profileState)
+  const bidirectionalBlocks = await maybeGetBlocksForReferenceList({
+    ctx,
+    params,
+    skeleton,
+    listState,
+  })
+  return [listState, profileState, { bidirectionalBlocks }].reduce(
+    mergeStates,
+    {} as HydrationState,
+  )
+}
+
+const noBlocks = (input: RulesFnInput<Context, Params, SkeletonState>) => {
+  const { skeleton, hydration } = input
+  if (!hydration.bidirectionalBlocks) {
+    return skeleton
+  }
+  skeleton.listitems = skeleton.listitems.filter(({ did }) => {
+    const blocks = hydration.bidirectionalBlocks?.get(did)?.values() || []
+    for (const block of blocks) {
+      if (block) return false
+    }
+    return true
+  })
```

Comment:
I very well may have this wrong, but I think the way hydration is working, we want something like this:
```suggestion
  const blocks = hydration.bidirectionalBlocks?.get(ownerDid)
  skeleton.listitems = skeleton.listitems.filter(({ did }) => {
    return blocks?.get(did)
  })
```

Reply:
(I believe `bidirectionalBlocks` is a map of maps `DID -> DID -> boolean`.)

Reply:
hmm let me try that again. i think i actually tried that but now curious again.

Reply:
argh, i wasn't building before test lol, sick.